resources/assets/coffee/react/wiki-search/suggestion.coffee

```diff
@@ -0,0 +1,69 @@
+###
+#    Copyright 2015-2018 ppy Pty. Ltd.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+{div, span} = ReactDOMFactories
+
+class @WikiSearch.Suggestion extends React.Component
+  onMouseEnter: (e) =>
+    $.publish 'suggestion:mouseenter', parseInt e.target.getAttribute 'position'
+
+  onMouseLeave: =>
+    $.publish 'suggestion:mouseleave'
+
+  onClick: (e) =>
+    $.publish 'suggestion:select', parseInt e.target.getAttribute 'position'
+
+  render: ->
+    queryString = $('.js-wiki-search-input').val()
+    titleSplit = @props.suggestion.title.split RegExp "(?=#{queryString})", 'i'
+
+    pathCleaned = @cleanPath @props.suggestion.path, @props.suggestion.title
+
+    div
+      className: osu.classWithModifiers 'wiki-search-suggestions__suggestion', ['highlighted' if @props.highlighted]
+      position: @props.position
+      onMouseEnter: @onMouseEnter
+      onMouseLeave: @onMouseLeave
+      onClick: @onClick
+      @formatTitle titleSplit, queryString
```

Comment:
text after the matching keyword gets truncated; e.g. `Some Tournament 2018` becomes `Some Tournament` when searching for `Tournament`, `Taiko Wars` truncated to `Taiko`, etc.
Use Elasticsearch to get the highlighting